pyproject.toml

```diff
@@ -151,6 +151,9 @@ confluent-kafka = ">=1.9.2,<3.0.0"
 portalocker = "^2.7.0"
 structlog = "^23.1.0"
 structlog-sentry = "^2.0.2"
+# pin dnspython to avoid dependency incompatibility
+dnspython = "2.3.0"
```

Comment:
Have you explored if changing the specification of `pymongo` in L299 to a version that locks dnspython to <2.4.2 works?

Reply:
If that pymongo version exists that is.

Reply:
Also do we need to backport to `3.6.x`?

Reply:
> If that pymongo version exists that is.

I didn't find any.

> Also do we need to backport to 3.6.x?

Yup, I'll do that once this is merged. I'll check 3.5.x and 3.4.x as well